httpx/_exceptions.py

```diff
@@ -71,6 +71,13 @@ def request(self) -> Request:
     def request(self, request: Request) -> None:
         self._request = request
 
+    def __reduce__(
+        self,
+    ) -> typing.Tuple[
+        typing.Callable[..., Exception], typing.Tuple[typing.Any], dict[str, typing.Any]
+    ]:
+        return (Exception.__new__, (type(self),) + self.args, self.__dict__)
+
```

Comment:
Are we able to be consistent with how pickle is supported on `Request`/`Response` here?

Reply:
All of the key word only arguments of `Request` have default values so that's why it's pickleable. Our choices here are:
1. Give all kwonly arguments defaults. Classes where all kwonly arguments have defaults mostly have the correct pickle behavior without any special support.
2. Do something like what I have here.

As an aside, this is a bit of a Python wart, ideally simple classes with required kwonly constructor args should be pickleable out of the box. I wonder if there is any discussion on discuss.python.org about it...

Reply:
Okay sure thing.
Thanks for the fix. ☺️